src/api/Metadata.js

```diff
@@ -926,7 +926,17 @@ class Metadata extends File {
                 const key = this.getMetadataCacheKey(id);
                 const cachedMetadata = cache.get(key);
 
-                const templateInstance = { ...template, type: metadata.data.$type };
+                let foundTemplate;
+                if (cachedMetadata && cachedMetadata.templates) {
+                    foundTemplate = cachedMetadata.templates.find(
+                        cachedTemplate =>
+                            cachedTemplate.templateKey === template.templateKey &&
+                            cachedTemplate.scope === template.scope,
+                    );
+                }
+                const templateId = foundTemplate ? foundTemplate.id : template.id;
```

Comment:
Can foundTemplate.id and template.id ever not be the same?

Reply:
It will in basically all the cases. Newly created template Instance (here defined as template) is getting assigned with random string as an id. To get id values consistent, we search for corresponding template and assign its id value for template instance. In rare case when corresponding cachedMetadata.templates do not exist, we would leave existing id as a fallback value.

Reply:
So if that's the case then what purpose does it serve to even do this?